crates/orchestrator/src/jobs/snos_job/mod.rs

```diff
@@ -35,28 +74,99 @@ impl Job for SnosJob {
         })
     }
 
-    async fn process_job(&self, _config: Arc<Config>, _job: &mut JobItem) -> Result<String, JobError> {
-        // 1. Fetch SNOS input data from Madara
-        // 2. Import SNOS in Rust and execute it with the input data
-        // 3. Store the received PIE in DB
-        todo!()
+    async fn process_job(&self, config: Arc<Config>, job: &mut JobItem) -> Result<String, JobError> {
+        let block_number = self.get_block_number_from_metadata(job)?;
+        let rpc_url = get_env_var_or_panic("MADARA_RPC_URL"); // should never panic at this point
```

Comment:
I checked the `starknet-rs` documentation and it does not look possible to retrieve the RPC url once it bas been "consumed"

Reply:
I see, we usually want to avoid reading env variables in the middle of the code and want them to be fetched during init. Can we create a new field inside the Config for madara rpc url in that case? wdyt?